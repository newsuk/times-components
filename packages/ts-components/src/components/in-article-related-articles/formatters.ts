import get from 'lodash.get';

export const formatRelatedArticles = (relatedArticleSlice: any) =>
  relatedArticleSlice.items.map(({ article }: any) => ({
    label: article.label,
    headline: article.shortHeadline || article.headline,
    link: article.url,
    summary:
      get(article, 'summary225[0].children[0].attributes.value') ||
      get(article, 'summary225[0].children[0].children[0].attributes.value'),
    publishedTime: article.publishedTime,
    image:
      get(article, 'leadAsset.crop169.url') ||
      get(article, 'leadAsset.posterImage.crop169.url'),
    byline: get(article, 'bylines[0].byline[0].children[0].attributes.value')
  }));
