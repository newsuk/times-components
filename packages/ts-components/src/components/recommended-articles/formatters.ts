import { SliceArticle } from '@times-components/ts-slices';

// TYPES

type Byline =
  | { __typename: string; type: 'author'; name: string }
  | { __typename: string; type: 'inline'; value: string };

type MediaCrop = {
  __typename: string;
  url: string;
  alt?: string;
  aspectRatio: string;
};
type Media = { __typename: string; crops: MediaCrop[] };

type SummaryText = { __typename: string; text: string };
type Summary = { __typename: string; children: SummaryText[] };

type Article = {
  __typename: string;
  url: string;
  slug: string;
  label: string;
  headline: string;
  publishedDateTime: string;
  bylines?: Byline[];
  summary?: Summary;
  media?: Media;
};

// HELPERS

const getBylines = (bylines?: Byline[]) =>
  bylines
    ? bylines
        .map(byline => (byline.type === 'author' ? byline.name : byline.value))
        .join('')
    : undefined;

const getSummary = (summary?: Summary) =>
  summary && summary.children
    ? summary.children.map(child => child.text).join('') + '...'
    : undefined;

const getImageCrops = (media?: Media) =>
  media && media.crops
    ? media.crops.map(crop => ({ url: crop.url, ratio: crop.aspectRatio }))
    : [];

// MAIN

const formatArticle = (article: Article): SliceArticle => {
  return {
    url: article.url,
    label: article.label,
    byline: getBylines(article.bylines),
    headline: article.headline,
    summary: getSummary(article.summary),
    datePublished: article.publishedDateTime,
    images: {
      alt: article.headline,
      crops: getImageCrops(article.media)
    }
  };
};

const getSliceName = (numOfArticles: number) => {
  switch (numOfArticles) {
    case 1:
      return 'RELATED_ARTICLE_1';
    case 2:
      return 'RELATED_ARTICLE_2';
    default:
      return 'RELATED_ARTICLE_3';
  }
};

export const getRecommendedArticlesSlice = (articles: Article[]) => {
  return {
    name: getSliceName(articles.length),
    children: articles
      .map((article: Article) => ({
        article: formatArticle(article)
      }))
      .slice(0, 3)
  };
};
