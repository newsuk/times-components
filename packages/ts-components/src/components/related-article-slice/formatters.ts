import { SliceArticle } from '@times-components/ts-slices';
import get from 'lodash.get';
import { Asset, RelatedArticle, RelatedArticles } from './types';

const decodeEntities = (str: string) =>
  str
    .replace(/&#(\d+);/g, (__: string, d: number) => String.fromCharCode(d))
    .replace(/\s+/g, ' ');

const isLabelEqualTo = (label: string, match?: string) =>
  label && match && label.toLowerCase() === match.toLowerCase();

const isLabelAttrValid = (title: string, label: string, byline?: string) =>
  !isLabelEqualTo(label, title) && !isLabelEqualTo(label, byline);

const getLabelAttributes = (label?: string) =>
  label ? label.split(' | ') : [];

const formatLabel = (title: string, article: Partial<SliceArticle>) =>
  getLabelAttributes(article.label)
    .map((a: string) => decodeEntities(a))
    .filter((a: string) => isLabelAttrValid(title, a, article.byline))
    .join(' | ');

const flattenAssets = (asset: Asset) => {
  const crops = [];
  if (asset.crop169) {
    crops.push({
      url: asset.crop169.url,
      ratio: '16:9'
    });
  }
  if (asset.crop32) {
    crops.push({
      url: asset.crop32.url,
      ratio: '3:2'
    });
  }
  return crops;
};
const formatArticle = (
  article: RelatedArticle,
  numberOfArticles: number
): SliceArticle => {
  const leadAssetCrops =
    (article.leadAsset &&
      article.leadAsset.posterImage &&
      flattenAssets(article.leadAsset.posterImage)) ||
    [];

  const leadAsset = {
    title: get(article, 'leadAsset.title'),
    crops: leadAssetCrops
  };
  const summary =
    numberOfArticles === 1 &&
    get(article, 'summary225[0].children[0].attributes.value');

  const byline = get(
    article,
    'bylines[0].byline[0].children[0].attributes.value'
  );

  const crops = (article.leadAsset && flattenAssets(article.leadAsset)) || [];

  return {
    id: article.id,
    url: article.url,
    label: formatLabel(leadAsset.title!, { label: article.label!, byline }),
    byline,
    headline: article.headline,
    summary,
    datePublished: article.publishedTime,
    dateUpdated: article.updatedTime,
    images: {
      alt: '',
      crops: [...crops, ...leadAssetCrops]
    }
  };
};

const getSliceName = (relatedArticleSlice: RelatedArticles) => {
  if (relatedArticleSlice.items.length === 1) {
    return 'RELATED_ARTICLE_1';
  }
  if (relatedArticleSlice.items.length === 2) {
    return 'LEAD_2';
  }
  return 'LEAD_1_AND_2';
};
const formatSlice = (relatedArticleSlice: RelatedArticles) => {
  return {
    name: getSliceName(relatedArticleSlice),
    children: relatedArticleSlice.items.map(relatedArticle => ({
      article: formatArticle(
        relatedArticle.article,
        relatedArticleSlice.items.length
      )
    }))
  };
};
export default formatSlice;
