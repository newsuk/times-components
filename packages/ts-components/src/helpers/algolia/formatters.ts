import { stripTags } from '@times-components/utils';
import get from 'lodash.get';
import { AlgoliaArticle, IndexedArticle } from './algoliaRelatedArticles';

const formatSummaryContent = (article: IndexedArticle, length: number) => {
  const summary = stripTags(article.content, ' ').trim();
  return summary.substring(0, summary.lastIndexOf(' ', length));
};

const formatSummary = (article: IndexedArticle, length: number) => [
  {
    children: [
      {
        attributes: { value: formatSummaryContent(article, length) },
        children: [],
        name: 'text'
      }
    ],
    name: 'paragraph'
  }
];

const formatImage = (article: AlgoliaArticle, ratio: string) =>
  get(article, 'leadAsset.crop.ratio') === ratio
    ? { url: get(article, 'leadAsset.crop.url') }
    : undefined;

export const formatArticle = (article: any) => ({
  article: {
    id: article.id,
    url: article.url,
    leadAsset: {
      id: get(article, 'leadAsset.id'),
      title: get(article, 'leadAsset.title'),
      crop169: formatImage(article, '16:9')
    },
    label: article.label,
    section: article.section,
    hasVideo: article.hasVideo,
    headline: article.headline,
    shortHeadline: article.shortHeadline,
    bylines: [{ byline: article.byline }],
    publishedTime: article.publishedTime,
    summary125: formatSummary(article, 125),
    summary145: formatSummary(article, 145)
  }
});

type Bylines = Array<{ byline: Array<{ children: any[] }> }>;

export const formatByLines = (bylines: Bylines) =>
  bylines &&
  bylines.length === 1 &&
  bylines[0].byline
    .map(({ children }) =>
      children.map(({ attributes }) => attributes.value).join(' ')
    )
    .join(' ');
