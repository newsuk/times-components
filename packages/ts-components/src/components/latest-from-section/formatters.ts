import { RelatedArticleType } from '../../types/related-article-slice';
type LatestSectionItem = {
  id: string;
  headline: string;
  canonicalID: string;
  summary: string;
  publishedTime: string;
  updatedTime: string;
  author: string;
  slug: string;
  label?: string;
  shortIdentifier: string;
  leadAsset: Array<{
    ratio: string;
    source: string;
  }>;
};

export type LatestSection = {
  section: string;
  items: LatestSectionItem[];
};

export const getSectionTitle = (section: LatestSection) => {
  return section && section.section;
};

export const formatLatestFromSection = (
  section: LatestSection
): RelatedArticleType[] | undefined => {
  return (
    section &&
    section.items.map<RelatedArticleType>((article: LatestSectionItem) => ({
      id: article.id,
      shortIdentifier: article.shortIdentifier,
      slug: article.slug,
      label: article.label || null,
      headline: article.headline,
      section: section.section,
      url: `/article/${article.slug}-${article.shortIdentifier}`,
      leadAsset: {
        crop169: {
          url:
            article.leadAsset &&
            article.leadAsset.find((crop: any) => crop.ratio === '16:9')!.source
        },
        crop32: {
          url:
            article.leadAsset &&
            article.leadAsset.find((crop: any) => crop.ratio === '3:2')!.source
        }
      },
      summary125: [
        {
          name: 'paragraph',
          children: [
            {
              name: 'text',
              attributes: {
                value: article.summary
              },
              children: []
            }
          ]
        }
      ],
      summary145: [
        {
          name: 'paragraph',
          children: [
            {
              name: 'text',
              attributes: {
                value: article.summary
              },
              children: []
            }
          ]
        }
      ],
      publishedTime: article.publishedTime,
      updatedTime: article.updatedTime,
      bylines: [
        {
          byline: [
            {
              name: 'author',
              children: [
                {
                  name: 'text',
                  children: [],
                  attributes: {
                    value: article.author
                  }
                }
              ],
              attributes: {
                slug: 'didi-tang'
              }
            }
          ],
          image: null
        }
      ]
    }))
  );
};
