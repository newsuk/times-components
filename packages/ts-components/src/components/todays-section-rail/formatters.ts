import { RelatedArticleType } from '../../types/related-article-slice';
type TodaysSectionItem = {
  id: string;
  headline: string;
  slug: string;
  url?: string;
  shortIdentifier?: string;
  leadAsset: Array<{
    ratio: string;
    source: string;
  }>;
  media?: {}
};

export type TodaysSection = {
  section: string;
  items: TodaysSectionItem[];
};

export const getSectionTitle = (section: TodaysSection) => {
  return section && section.section;
};

export const formatTodaysSection = (
  section: TodaysSection
): RelatedArticleType[] | undefined => {
  console.log('XXXXXXXXXXXXXXXXXXXXXXXXXXXXX',section)
  return (
    section &&
    section.items.map<RelatedArticleType>((article: TodaysSectionItem) => ({
      id: article.id,
      slug: article.slug,
      label: null,
      shortIdentifier: article.shortIdentifier,
      headline: article.headline,
      section: section.section,
      url: article.url ? article.url :`/article/${article.slug}-${article.shortIdentifier}`,
      leadAsset: {
        crop169: {
          url:
            article.leadAsset &&
            article.leadAsset.find((crop: any) => crop.ratio === '16:9')!.source
        },
        crop32: {
          url:
            article.leadAsset &&
            article.leadAsset.find((crop: any) => crop.ratio === '3:2')!.source
        }
      },
      bylines: []
    }))
  );
};
