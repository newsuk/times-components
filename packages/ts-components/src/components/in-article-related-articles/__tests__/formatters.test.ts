import { formatRelatedArticles } from '../formatters';
const sample = {
  sliceName: 'StandardSlice',
  items: [
    {
      leadAsset: null,
      article: {
        leadAsset: {
          crop169: {
            url:
              'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fb11686ce-f43d-11eb-8f01-2c678acbb979.jpg?crop=4529%2C2548%2C2%2C450'
          },
          crop32: {
            url:
              'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fb11686ce-f43d-11eb-8f01-2c678acbb979.jpg?crop=4552%2C3035%2C0%2C0'
          },
          id: '39de6425-613f-42d9-847b-cd49b16d7e65',
          title: "Tokyo Olympics. Day 11. Cycling Track - Women's Team Pursuit"
        },
        bylines: [
          {
            byline: [
              {
                name: 'author',
                children: [
                  {
                    name: 'text',
                    children: [],
                    attributes: {
                      value: 'Rebecca Myers'
                    }
                  }
                ],
                attributes: {
                  slug: 'rebecca-myers'
                }
              }
            ],
            image: null
          },
          {
            byline: [
              {
                name: 'inline',
                children: [
                  {
                    name: 'text',
                    children: [],
                    attributes: {
                      value: ', Tokyo'
                    }
                  }
                ]
              }
            ],
            image: null
          }
        ],
        hasVideo: false,
        headline:
          'Tokyo Olympics: Super-fast rivals force Jason and Laura Kenny to settle for silver',
        id: 'fc79fad0-f43b-11eb-8f01-2c678acbb979',
        label: 'Olympics',
        publicationName: 'TIMES',
        publishedTime: '2021-08-03T16:00:00.000Z',
        updatedTime: '2021-08-03T22:51:28.000Z',
        section: 'sport',
        shortIdentifier: 'lfjbwnpw2',
        slug:
          'tokyo-olympics-jason-kenny-and-laura-kenny-have-to-settle-for-silver-as-records-tumble',
        url:
          'https://www.thetimes.co.uk/article/tokyo-olympics-jason-kenny-and-laura-kenny-have-to-settle-for-silver-as-records-tumble-lfjbwnpw2',
        summary105: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were'
                },
                children: []
              }
            ]
          }
        ],
        summary125: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by'
                },
                children: []
              }
            ]
          }
        ],
        summary145: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by super-fast rivals during the'
                },
                children: []
              }
            ]
          }
        ],
        summary160: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by super-fast rivals during the defences of'
                },
                children: []
              }
            ]
          }
        ],
        summary175: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by super-fast rivals during the defences of their Olympic'
                },
                children: []
              }
            ]
          }
        ],
        summary225: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by super-fast rivals during the defences of their Olympic titles. They took second place in the sprint and team'
                },
                children: []
              }
            ]
          }
        ]
      }
    },
    {
      leadAsset: null,
      article: {
        leadAsset: {
          posterImage: {
            crop169: {
              url:
                'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fc2046bdc-f46e-11eb-9beb-a8c8694a90e3.jpg?crop=1278%2C719%2C0%2C0'
            },
            crop32: {
              url:
                'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fc2046bdc-f46e-11eb-9beb-a8c8694a90e3.jpg?crop=1078%2C719%2C100%2C0'
            },
            id: 'f874d69b-bd32-4734-b1f0-37f9cd36dec9',
            title: ''
          }
        },
        bylines: [
          {
            byline: [
              {
                name: 'inline',
                children: [
                  {
                    name: 'text',
                    children: [],
                    attributes: {
                      value: 'Ed Gorman, George Sandeman'
                    }
                  }
                ]
              }
            ],
            image: null
          }
        ],
        hasVideo: true,
        headline:
          'Team GB: Keely Hodgkinson wins 800m silver, breaking Kelly Holmes’s record',
        id: 'd84e14e2-f41e-11eb-a2a3-afea84050239',
        label: 'Olympics',
        publicationName: 'TIMES',
        publishedTime: '2021-08-03T13:50:00.000Z',
        updatedTime: '2021-08-03T16:07:58.000Z',
        section: 'sport',
        shortIdentifier: 'vxc7d03f8',
        shortHeadline:
          'Keely Hodgkinson wins 800m silver, breaking Kelly Holmes’s record',
        slug:
          'tokyo-olympics-britain-steal-gold-from-new-zealand-in-49er-class-race-before-giles-scott-triumphs-in-finn-class',
        url:
          'https://www.thetimes.co.uk/article/tokyo-olympics-britain-steal-gold-from-new-zealand-in-49er-class-race-before-giles-scott-triumphs-in-finn-class-vxc7d03f8',
        summary105: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame'
                },
                children: []
              }
            ]
          }
        ],
        summary125: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s'
                },
                children: []
              }
            ]
          }
        ],
        summary145: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s national record to clinch'
                },
                children: []
              }
            ]
          }
        ],
        summary160: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s national record to clinch silver in the'
                },
                children: []
              }
            ]
          }
        ],
        summary175: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s national record to clinch silver in the 800m, making the'
                },
                children: []
              }
            ]
          }
        ],
        summary225: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'text',
                attributes: {
                  value:
                    'Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s national record to clinch silver in the 800m, making the 11th day of the Games one of the nation’s most'
                },
                children: []
              }
            ]
          }
        ]
      }
    },
    {
      leadAsset: null,
      article: {
        leadAsset: {
          crop169: {
            url:
              'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F45c33d56-20f7-11ea-9a1b-70ae6678484b.jpg?crop=1493%2C840%2C4%2C152'
          },
          crop32: {
            url:
              'https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F45c33d56-20f7-11ea-9a1b-70ae6678484b.jpg?crop=1500%2C1000%2C0%2C0'
          },
          id: '7781b9f1-6623-4700-9618-d012e2a40028',
          title: ''
        },
        bylines: [
          {
            byline: [
              {
                name: 'inline',
                children: [
                  {
                    name: 'text',
                    children: [],
                    attributes: {
                      value: 'Jane Edwardes and Fiona Mountford'
                    }
                  }
                ]
              }
            ],
            image: null
          }
        ],
        hasVideo: false,
        headline: 'Theatre, also showing',
        id: '7ddc2df0-20d0-11ea-81b4-b78674dd3224',
        label: 'CHRISTMAS THEATRE',
        publicationName: 'SUNDAYTIMES',
        publishedTime: '2019-12-22T00:01:00.000Z',
        updatedTime: '2019-12-22T18:04:30.000Z',
        section: 'culture',
        shortIdentifier: 'pldxw3prv',
        shortHeadline: 'Theatre, also showing',
        slug: 'theatre-also-showing',
        url:
          'https://www.thetimes.co.uk/article/theatre-also-showing-pldxw3prv',
        summary105: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while'
                },
                children: []
              }
            ]
          }
        ],
        summary125: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while Frank Loesser’s classic'
                },
                children: []
              }
            ]
          }
        ],
        summary145: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while Frank Loesser’s classic isn’t as radical as'
                },
                children: []
              }
            ]
          }
        ],
        summary160: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while Frank Loesser’s classic isn’t as radical as some of its'
                },
                children: []
              }
            ]
          }
        ],
        summary175: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while Frank Loesser’s classic isn’t as radical as some of its previous shows, it'
                },
                children: []
              }
            ]
          }
        ],
        summary225: [
          {
            name: 'paragraph',
            children: [
              {
                name: 'bold',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Guys and Dolls'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'italic',
                children: [
                  {
                    name: 'text',
                    attributes: {
                      value: 'Crucible, Sheffield'
                    },
                    children: []
                  }
                ]
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value: '★★★★'
                },
                children: []
              },
              {
                name: 'break',
                children: []
              },
              {
                name: 'text',
                attributes: {
                  value:
                    'The Crucible has a history of staging great musicals, and while Frank Loesser’s classic isn’t as radical as some of its previous shows, it still hits the spot. Broadway’s lowlife of'
                },
                children: []
              }
            ]
          }
        ]
      }
    }
  ]
};

describe('formatRelatedArticles', () => {
  it('formats', () => expect(formatRelatedArticles(sample)).toMatchSnapshot());
});
