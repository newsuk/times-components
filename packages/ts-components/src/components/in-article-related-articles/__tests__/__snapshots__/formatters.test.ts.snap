// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formatRelatedArticles formats 1`] = `
Array [
  Object {
    "byline": "Rebecca Myers",
    "headline": "Tokyo Olympics: Super-fast rivals force Jason and Laura Kenny to settle for silver",
    "image": "https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fb11686ce-f43d-11eb-8f01-2c678acbb979.jpg?crop=4529%2C2548%2C2%2C450",
    "label": "Olympics",
    "link": "https://www.thetimes.co.uk/article/tokyo-olympics-jason-kenny-and-laura-kenny-have-to-settle-for-silver-as-records-tumble-lfjbwnpw2",
    "publishedTime": "2021-08-03T16:00:00.000Z",
    "summary": "It was a night of silver linings for Jason and Laura Kenny after the power couple of British Cycling were stunned by super-fast rivals during the defences of their Olympic titles. They took second place in the sprint and team",
  },
  Object {
    "byline": "Ed Gorman, George Sandeman",
    "headline": "Keely Hodgkinson wins 800m silver, breaking Kelly Holmes’s record",
    "image": "https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Ftimes%2Fprod%2Fweb%2Fbin%2Fc2046bdc-f46e-11eb-9beb-a8c8694a90e3.jpg?crop=1278%2C719%2C0%2C0",
    "label": "Olympics",
    "link": "https://www.thetimes.co.uk/article/tokyo-olympics-britain-steal-gold-from-new-zealand-in-49er-class-race-before-giles-scott-triumphs-in-finn-class-vxc7d03f8",
    "publishedTime": "2021-08-03T13:50:00.000Z",
    "summary": "Great Britain celebrated double Olympic gold in sailing this morning, while Keely Hodgkinson broke Dame Kelly Holmes’s national record to clinch silver in the 800m, making the 11th day of the Games one of the nation’s most",
  },
  Object {
    "byline": "Jane Edwardes and Fiona Mountford",
    "headline": "Theatre, also showing",
    "image": "https://www.thetimes.co.uk/imageserver/image/%2Fmethode%2Fsundaytimes%2Fprod%2Fweb%2Fbin%2F45c33d56-20f7-11ea-9a1b-70ae6678484b.jpg?crop=1493%2C840%2C4%2C152",
    "label": "CHRISTMAS THEATRE",
    "link": "https://www.thetimes.co.uk/article/theatre-also-showing-pldxw3prv",
    "publishedTime": "2019-12-22T00:01:00.000Z",
    "summary": "Guys and Dolls",
  },
]
`;
